openfisca_core/holders.py

```diff
@@ -112,164 +110,87 @@ def clone(self):
         return new
 
     def compute(self, period = None, **parameters):
-        """Compute array if needed and/or convert it to requested period and return a dated holder containig it.
+        """Compute array if needed and return a dated holder containing it.
 
         The returned dated holder is always of the requested period and this method never returns None.
         """
         if period is None:
             period = self.simulation.period
         column = self.column
-        accept_other_period = parameters.get('accept_other_period', False)
 
-        # First look for dated_holders covering the whole period (without hole).
-        dated_holder = self.get_from_cache(period, parameters.get('extra_params'))
-        if dated_holder.array is not None:
-            return dated_holder
+        # Check that the requested period matches period_behavior
+        if not column.period_behavior == PERMANENT:
+            if ((column.period_behavior == MONTH and period.unit != periods.MONTH) or
+                    (column.period_behavior == YEAR and period.unit != periods.YEAR)):
+                raise ValueError('Computation requested with wrong period unit for variable {} ({} instead of {})'.format(
```

Comment:
A link to the documentation wouldn't be even better ?

`Unable to compute monthly variable {} for year-long period {}. See https://doc.openfisca.fr/coding-the-legislation/35_periods.html'`